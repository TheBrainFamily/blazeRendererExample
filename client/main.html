<head>
  <title>todos</title>
</head>

<body>
  <h1>Welcome to Meteor!</h1>

{{> main }}
</body>

<template name="main">

    {{> hello}}
    {{> info}}
    {{> developers }}
</template>
<template name="hello">
  <button>Click Me</button>
    <div>
  <p>You've pressed the button {{counter}} times.</p>
    </div>
</template>

<template name="info">
  <h2>Learn Meteor!</h2>
  <ul class="test-meteor-links">
    <li><a href="https://www.meteor.com/try" target="_blank">Do the Tutorial</a></li>
    <li><a href="http://guide.meteor.com" target="_blank">Follow the Guide</a></li>
  </ul>
</template>

<template name="developers">
    <div class="test-developers">
        {{#if developers.length}}
            <li>
            {{#each developers}}
                <ul>{{name}} {{surname}} @{{company}}</ul>
                {{/each}}
            </li>
        {{else}}
            <p>No developers :-(</p>
        {{/if}}
    </div>
</template>