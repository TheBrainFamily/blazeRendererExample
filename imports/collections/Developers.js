import { Mongo } from "meteor/mongo";

const Developers = new Mongo.Collection("Players");
export { Developers };
