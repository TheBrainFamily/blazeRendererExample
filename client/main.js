import { Template } from "meteor/templating";
import { ReactiveVar } from "meteor/reactive-var";
import { Developers } from "../imports/collections/Developers";
import "./main.html";

Template.hello.onCreated(function helloOnCreated() {
  // counter starts at 0
  this.counter = new ReactiveVar(0);
});

Template.hello.helpers({
  counter() {
    return Template.instance().counter.get();
  }
});

Template.hello.events({
  "click button": function(event, instance) {
    // increment the counter when button is clicked
    instance.counter.set(instance.counter.get() + 1);
  }
});

Template.developers.helpers({
  developers() {
    return Developers.find({});
  }
});
