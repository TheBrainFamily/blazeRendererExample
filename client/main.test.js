import renderBlaze from 'blaze-renderer'
import cheerio from 'cheerio'
import './main.js';
import {Developers} from "../imports/collections/Developers"
import { runInFiber } from 'testable-meteor'

test('renders main', runInFiber(() => {
  expect(renderBlaze('main')).toMatchSnapshot()
}))

test('it shows no developers info when they are not available', runInFiber(() => {
  const $ = cheerio.load(renderBlaze('main'))
  expect($('.test-developers p').html()).toEqual("No developers :-(")
}))

test('it displays the developers when available', runInFiber(() => {
  Developers.insert({name: "Lukasz", surname: "Gandecki", company: "TheBrain Software House"})
  Developers.insert({name: "Sam", surname: "Hatoum", company: "Xolv.io"})

  const $ = cheerio.load(renderBlaze('main'))

  expect($('.test-developers li ul').length).toEqual(2)
  expect($('.test-developers li ul:nth-child(1)').html()).toEqual("Lukasz Gandecki @TheBrain Software House")
  expect($('.test-developers li ul:nth-child(2)').html()).toEqual("Sam Hatoum @Xolv.io")
}))