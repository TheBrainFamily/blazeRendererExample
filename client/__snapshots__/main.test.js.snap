// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main 1`] = `
"<button>Click Me</button>
<div>
    <p>You've pressed the button 0 times.</p>
</div>
<h2>Learn Meteor!</h2>
<ul class=\\"test-meteor-links\\">
    <li><a href=\\"https://www.meteor.com/try\\" target=\\"_blank\\">Do the Tutorial</a></li>
    <li><a href=\\"http://guide.meteor.com\\" target=\\"_blank\\">Follow the Guide</a></li>
</ul>
<div class=\\"test-developers\\">

    <p>No developers :-(</p>

</div>"
`;
